marlowe-playground-client/src/MonadApp.purs

```diff
@@ -239,57 +189,22 @@ withHaskellEditor ::
   HalogenApp m (Maybe a)
 withHaskellEditor = HalogenApp <<< Editor.withEditor _haskellEditorSlot unit
 
-withMarloweEditor ::
-  forall m a.
-  MonadEffect m =>
-  (Editor -> m a) ->
-  HalogenApp m (Maybe a)
-withMarloweEditor = HalogenApp <<< Editor.withEditor _marloweEditorSlot unit
-
 updateContractInStateP :: String -> MarloweState -> MarloweState
 updateContractInStateP text state = case parseContract text of
   Right parsedContract ->
     let
       lintResult = lint parsedContract
 
-      warnings =
-        map (warningToAnnotation text "The contract can make a negative payment here") (view L._negativePayments lintResult)
-          <> map (warningToAnnotation text "The contract can make a negative deposit here") (view L._negativeDeposits lintResult)
-          <> Set.toUnfoldable (Set.map (warningToAnnotation text "Timeouts should always increase in value") (view L._timeoutNotIncreasing lintResult))
-          <> map (warningToAnnotation text "The contract tries to Use a ValueId that has not been defined in a Let") (view L._uninitializedUse lintResult)
-          <> map (warningToAnnotation text "Let is redefining a ValueId that already exists") (view L._shadowedLet lintResult)
-          <> map (warningToAnnotation text "This Observation will always evaluate to True") (view L._trueObservation lintResult)
-          <> map (warningToAnnotation text "This Observation will always evaluate to False") (view L._falseObservation lintResult)
-
       mContract = fromTerm parsedContract
     in
       case mContract of
         Just contract -> do
-          set _editorWarnings warnings <<< set _editorErrors [] <<< set _contract (Just contract) $ state
+          set _editorErrors [] <<< set _contract (Just contract) $ state
         Nothing -> do
           let
             (Holes holes) = view L._holes lintResult
-
-            holesArray = Set.toUnfoldable $ fold $ Map.values holes
-
-            errors = map (holeToAnnotation text) holesArray
-          (set _editorWarnings warnings <<< set _editorErrors errors <<< set _holes (Holes holes)) state
-  Left error -> (set _editorErrors [ errorToAnnotation text error ] <<< set _holes mempty) state
-
-warningToAnnotation :: String -> String -> Position -> Annotation
-warningToAnnotation str text { row, column } = { column, row, text, "type": "warning" }
-
-holeToAnnotation :: String -> MarloweHole -> Annotation
-holeToAnnotation str (MarloweHole { name, marloweType, row, column }) = { column, row, text: "Found hole ?" <> name, "type": "warning" }
-
-errorToAnnotation :: String -> ContractParseError -> Annotation
-errorToAnnotation str EmptyInput = { column: 0, row: 0, text: "No input provided", "type": "error" }
-
-errorToAnnotation str (ContractParseError { row, column, message, token }) =
-  let
-    msg = unlines $ take 4 $ lines message
-  in
-    { column, row, text: msg, "type": "error" }
+          (set _holes (Holes holes)) state
```

Comment:
I'm confused - why do you unwrap `holes` only to rewrap it?

Reply:
I have no idea